docs/lang/attributes.md

```diff
@@ -199,18 +199,43 @@ When we have following two conditions:
 1. An input port is marked with `@data` in the component definitions, and
 2. The cell instance is marked as `@data`
 
-The backend generate `'x` as the default value for the assignment to the port instead of `'0`. Additionally, if the port has exactly one assignment, the backend removes the guard entirely and produces a continuous assignment.
+The backend generates `'x` as the default value for the assignment to the port instead of `'0`. Additionally, if the port has exactly one assignment, the backend removes the guard entirely and produces a continuous assignment.
 
 This represents the optimization:
 ```
-in = g ? out : 'x
+cells {
+  @data cell = cell_component();
+}
+wires {
+  cell.in = g ? out : 'x
+}
 ```
 into:
 ```
-in = out;
+wires {
+  cell.in = out;
+}
 ```
 Since the value `'x` can be replaced with anything.
 
+`@data` can also appear on the output ports of a Calyx component.
+This represents when we want to optimize assignments to the port _within a Calyx component_.
+Consider the following example:
+```
+component main() ->(...@data out_write_data: 32....) {
+  ...
+  wires {
+    out_write_data = g ? out : 'x // we want to optimize to out_write_data = out
```

Comment:
This is great! Obviously, we don't support these "complete assignments" yet but it would be great to do that in the future so that we can represent the optimization within the compiler instead of inside the backend.

Reply:
Ah yeah, that's a good point. I added a sentence in the docs clarifying that all these optimizations occur during the calyx->verilog translation, and that calyx doesn't support these types of complete assignments.